### PR TITLE
vtk-m: Major refactor of the recipe

### PR DESCRIPTION
**Pull request by vicentebolea**
- It moves conflicts from configure time to concretization time
- It leverages in new Spack CMakePackage methods to specify CMake options in a way that significantly reduce the LoC.

### CODE REVIEW

**Review by alalazo — DISMISSED** [review]
Much cleaner. Waiting for reviewer approval before merging.

File: var/spack/repos/builtin/packages/vtk-m/package.py [hunk 1]
```diff
@@ -112,6 +112,12 @@ class VtkM(CMakePackage, CudaPackage, ROCmPackage):
 
     depends_on("hip@3.7:", when="+rocm")
 
+    conflicts("+ascent_types", when="+64bitids", msg="Ascent types requires 32 bit IDs")
+    conflicts("+logging", when="@:1.3.0", msg="logging is not supported in vtk-m<1.3")
+    conflicts("+mpi", when="@:1.3.0", msg="mpi is not supported in vtk-m<1.3")
+    conflicts("+openmp", when="@:1.3.0", msg="mpi is not supported in vtk-m<1.3")
+    conflicts("+testlib", when="@:1.7.0", msg="testlib is not supported in vtk-m<1.7")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/vtk-m/package.py]: Most of these (all maybe?) can be incorporated into the corresponding variants as a when clause.

**Review by alalazo — DISMISSED** [review]

**Comment by alalazo** [comment]
Can you solve the conflict?

**Comment by vicentebolea** [comment]
> Can you solve the conflict?

Done

**Comment by vicentebolea** [comment]
@alalazo ci passed :)

**Review by alalazo — APPROVED** [review]

**Pull request merged**